test/e2e/cypress/integration/clusters_overview.js

```diff
@@ -0,0 +1,306 @@
+        ['env1', 1],
+        ['env2', 1],
+        ['env3', 1],
+      ];
+      tagsScenarios.forEach(([tag, expectedTaggedClusters], index) => {
+        it(`should have ${expectedTaggedClusters} clusters tagged with tag '${tag}'`, () => {
+          cy.intercept('GET', `/clusters?per_page=100&tags=${tag}`).as(
+            'filterByTags'
+          );
+          const selectedOption = `#bs-select-5-${index}`;
+          cy.get(selectedOption).click();
+          cy.wait('@filterByTags').then(() => {
+            cy.get('.tn-clustername')
+              .its('length')
+              .should('eq', expectedTaggedClusters);
+            cy.get('.pagination-count').should(
+              'contain',
+              `${expectedTaggedClusters} items`
+            );
+            cy.get('.page-item')
+              .its('length')
+              .should('eq', Math.ceil(expectedTaggedClusters / 100) + 2);
+            resetFilter(selectedOption);
+          });
+        });
+      });
+    });
+
+    describe('Removing filtered tags', () => {
+      const tag = 'tag1';
+
+      before(() => {
+        // Wait for the POST that gets triggered on tag submission
+        cy.intercept('/api/clusters/**').as('tagPosted');
+        // Wait for the filter reset after the first POST finishes
+        cy.intercept('GET', '/api/tags?resource_type=clusters').as(
+          'filterRefreshed'
+        );
+
+        // Clear crap
+        cy.get('h1').click();
+
+        // We click on the tag button to open the tag modal in the first cluster
+        // and submit the tag
+        cy.get(
+          `#cluster-${availableClustersId[0]} > .tn-cluster-tags > .tagify`
+        )
+          .type(tag)
+          .click();
+
+        // We wait for the POST to finish and the filter to be refreshed
+        cy.wait('@tagPosted');
+        cy.wait('@filterRefreshed');
+
+        // We click on the tag button to open the tag modal in the second cluster
+        // and submit the tag
+        cy.get(
+          `#cluster-${availableClustersId[1]} > .tn-cluster-tags > .tagify`
+        )
+          .type(tag)
+          .click();
+
+        cy.wait('@tagPosted');
+        cy.wait('@filterRefreshed');
+        cy.get('.dropdown-item').contains(tag);
+        cy.get('.tn-filters > :nth-child(6) > .btn').click();
+      });
+
+      it(`should reload the clusters table when filtered tags are removed`, () => {
+        cy.intercept('GET', `/clusters?per_page=100&tags=${tag}`).as(
+          'filterByTags'
+        );
+        cy.get('.dropdown-item').contains(tag).click();
+        cy.wait('@filterByTags').then(() => {
+          cy.get('.tn-clustername').should('have.length', 2);
+        });
+
+        cy.get('.tn-filters > :nth-child(6) > .btn').click();
+
+        cy.intercept('DELETE', `${tag}`).as('firstTagRemoved');
+        cy.get(
+          `#cluster-${availableClustersId[0]} > .tn-cluster-tags > .tagify > .tagify__tag > .tagify__tag__removeBtn`
```

Comment:
I think we have a `x` class for the removing element. We could use it, instead of this long class queue

Reply:
Don't ask me why, I tried several times using this method (the same method used in the hosts overview) but without any luck :/